File: C++/027. Remove Element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int size=nums.size();
        int j=size-1;
        int i=0;
        if (size==0)
        {
            return 0;
        }
        while (j-i>=0)
        {
            if(nums[i]==val)
            {
                nums[i]=nums[j];
                --j;
            }
            else
            {
                ++i;
            }
        }
        return i;
    }
};

File: python/027. Remove Element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        n = len(nums)
        idx = 0
        for i in range(n):
            if nums[i] != val:
                nums[idx] = nums[i]
                idx += 1
        return idx